### PR TITLE
Redis IP validation for FQDN - Closes #868

### LINKED ISSUE #868: Allow setting redis host to hostname
One should be able to set [.redis.host](https://github.com/LiskHQ/lisk/blob/f82e286ae21f9a6df2f7905b9bb6261b8cb91ee3/config.json#L28) in `config.json` to a hostname or a FQDN.
This is not the case however and the application fails to start with the following error:

    Failed to validate config data [ { code: 'INVALID_FORMAT',
       params: [ 'ip', 'redis' ],
       message: 'Object didn\'t pass validation for format ip: redis',
       path: '#/redis/host',
       schemaId: 'appCon' } ]

### PR DESCRIPTION
**Pull request by nazarhussain**
Closes issue: #868

### CODE REVIEW

File: helpers/z_schema.js [hunk 1]
```diff
@@ -105,6 +106,13 @@ z_schema.registerFormat('ip', function (str) {
 	return ip.isV4Format(str);
 });
 
+/* Since an IP is not considered to be a hostname
```
Comment (diego-G) [inline helpers/z_schema.js]: We can remove this comment and change the header of the file.

Comment (nazarhussain) [reply helpers/z_schema.js]: @diego-G Done.

File: helpers/z_schema.js [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 var ip = require('ip');
 /**
- * Uses JSON Schema validator z_schema to register custom formats.
+ * Uses JSON Schema validator z_schema to register custom formats. <br/>
```
Comment (Isabello) [inline helpers/z_schema.js]: <br/> tag needed here?

File: test/unit/helpers/z_schema.js [hunk 3]
```diff
@@ -87,4 +87,26 @@ describe('schema - custom formats', function () {
...
+		it('should return false if value is not an IP or not a FQDN', function () {
+			var invalidData = ['192.168', 'alpha-', 'apha_server', 'alpha.server.'];
+
+			invalidData.forEach(function (item) {
+				expect(validator.validate(item, schema)).to.equal(false);
+			});			
+		});
+
+		it('should return true if value is an IP or a valid FQDN', function () {
+			var validData = ['192.168.0.1', '127.0.0.1', 'localhost', 'app.server', 'alpha.server.com'];
```
Comment (MaciejBaj) [inline test/unit/helpers/z_schema.js]: Add maybe a public IP here as well

File: helpers/z_schema.js (hunk 2 above)
Comment (nazarhussain) [reply helpers/z_schema.js]: Wanted to show next line to a next line in generated docs. That was the only way I found to do it.

File: test/unit/helpers/z_schema.js (hunk 3 above)
Comment (nazarhussain) [reply test/unit/helpers/z_schema.js]: @MaciejBaj  Done

File: helpers/z_schema.js (hunk 2 above)
Comment (Isabello) [reply helpers/z_schema.js]: Ok fine with me

**Comment by nazarhussain** [comment]
@diego-G Its not a new test file. Added few test cases into existing helpers test file. Which is already in the list at `index.js`

**Pull request merged**